### PR TITLE
New management overview page and rename stack management to dashboard management

### PR DESCRIPTION
**Pull request by Hailong-am**
### Description

Support navigation changes for administrative features. This change includes 
1. Rename stack management to Dashboard management
2. Add new management overview page
3. Replace Stack Management to Management overview page on home app categories page

### Issues Resolved

https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4132

#### Navigation and overview page
image

#### App category page
image

### Check List

- [ ] All tests pass
  - [x] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [x] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by manasvinibs** [comment]
@Hailong-am  Is this change targeted to 2.9 release? @KrooshalUX Tagging Kroosh for UX review on this.

**Comment by KrooshalUX** [comment]
@manasvinibs thanks for the ping. This work belongs to @kamingleung  - however I have UX feedback that I will work with him on directly.

**Review by BSFishy — APPROVED** [review]
Looks good on Oui usage

**Comment by manasvinibs** [comment]
> @manasvinibs thanks for the ping. This work belongs to @kamingleung - however I have UX feedback that I will work with him on directly.

@KrooshalUX Do you think we should put a block on this change till it gets UX reviewed?

**Comment by kamingleung** [comment]
@manasvinibs @Hailong-am Let's put a block on this until the UX feedback is addressed. Thanks.

**Comment by KrooshalUX** [comment]
A few notes from the Design System perspective for the Management page: 
- This page should follow the Directory pattern, where the page title would be on the page background, not on a panel. 
- The contents such as Index Management, Snapshot Management, Notifications, should be children of OuiCard, not OuiPanel. 
- For the cards, please refer to the "no icon" example in the documentation https://oui.opensearch.org/1.2/#/display/card
- Additionally, OuiCard does not utilize horizontal rules between 'title' and 'description'

**Review by manasvinibs — CHANGES_REQUESTED** [review]
Blocking this change till it gets reviewed by UX

**Comment by kamingleung** [comment]
@manasvinibs @KrooshalUX 
We have addressed the UX feedback with a fit-and-finish review. The changes have been updated to this PR, we can unblock and continue with this PR. Thanks.

image

**Review by manasvinibs — COMMENTED** [review]
Unblocking as changes are approved by UX review

**Review by manasvinibs — DISMISSED** [review]

**Comment by Hailong-am** [comment]
@manasvinibs can you help to add backport 2.x label, it is target for 2.9 release.

**Comment by Hailong-am** [comment]
Link checker will fixed by this PR https://github.com/opensearch-project/OpenSearch-Dashboards/pull/4384

**Comment by Hailong-am** [comment]
@ashwin-pc @manasvinibs can you help to do the review?

**Comment by Hailong-am** [comment]
Updates for stack management image

File: src/plugins/management_overview/public/plugin.ts
Comment (xluo-aws) [inline src/plugins/management_overview/public/plugin.ts:61]: Do we still show stack?

Comment (Hailong-am) [reply src/plugins/management_overview/public/plugin.ts:61]: Yes, based on mock up of @kamingleung

**Review by ashwin-pc — COMMENTED** [review]
The change looks mostly good. Have some concerns around how the management overview plugin exposes its API's. We should only expose the register method in the start contract.

File: CHANGELOG.md
Comment (Hailong-am) [reply CHANGELOG.md:91]: sure, changed to unreleased section.

File: src/plugins/management_overview/public/plugin.ts
Comment (Hailong-am) [reply src/plugins/management_overview/public/plugin.ts:95]: The reason is that for `dashboards management` and `dev_tools` they may have no sub application installed. For example, in case console is not installed, dev tools will update to hidden and not show up in left navigation.  the logic is placed in start phase, so to avoid register to management overview page at setup phase and then remove it if it's hidden, then i put the register the into start phase also.

Comment (Hailong-am) [reply src/plugins/management_overview/public/plugin.ts:95]: Let's me follow the pattern and limit the registration only into setup phase

File: src/plugins/management_overview/public/plugin.ts
Comment (Hailong-am) [reply src/plugins/management_overview/public/plugin.ts]: remove this function and limit the register only into setup lifecycle.

File: src/plugins/dev_tools/public/plugin.ts
Comment (Hailong-am) [reply src/plugins/dev_tools/public/plugin.ts:86]: If we keep this 9010 unchanged, we need to change dashboards management order to a lower value than 9010. Anyway we need to change order from one of them.

**Comment by Hailong-am** [comment]
Link checker is a temporary issue, re-run should able to pass

**Comment by Hailong-am** [comment]
> The change looks mostly good. Have some concerns around how the management overview plugin exposes its API's. We should only expose the register method in the start contract.

@ashwin-pc  remove the register in start lifecycle and limit to setup, please help to take a look

**Comment by manasvinibs** [comment]
@Hailong-am I see few functional test failures related to management settings. Can you look into it?

File: src/plugins/management_overview/public/application.tsx
Comment (ashwin-pc) [inline src/plugins/management_overview/public/application.tsx:24]: Nice! this is much better :)

**Review by ashwin-pc — COMMENTED** [review]
@Hailong-am Thanks, the changes look good. Looks like you need to update the functional test however, can you fix those too? the PR should be good to go after that.

**Comment by Hailong-am** [comment]
> @Hailong-am I see few functional test failures related to management settings. Can you look into it?

@manasvinibs @ashwin-pc all tests are passed.

**Review by manasvinibs — APPROVED** [review]

**Comment by Hailong-am** [comment]
@ashwin-pc can we get this merged?

**Review by ashwin-pc — APPROVED** [review]
This looks good to me. You are also adding functional tests for the new page right?

**Review by ananzh — APPROVED** [review]

**Comment by Hailong-am** [comment]
> This looks good to me. You are also adding functional tests for the new page right?

sure, will add functional tests in separate pull request.

**Pull request merged**
